File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/GameEndedEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class GameEndedEvent(
        val numberOfRounds: Int,
        val results: Set<BotResults>
) : Content(type = ContentType.BOT_RESULTS.type)

File: bot-api/python/src/robocode_tank_royale/bot_api/events/game_ended_event.py
from ..bot_results import BotResults
from .event_abc import EventABC


class GameEndedEvent(EventABC):
    """
    Event triggered when the game ends.

    Attributes:
        number_of_rounds (int): The total number of rounds played in the game.
        results (BotResults): The results of the battle for this bot, containing detailed scores and performance metrics.
    """

    number_of_rounds: int
    results: BotResults
